.github/workflows/workflow-path-to-live.yml

```diff
@@ -0,0 +1,347 @@
+      - integration_tests_admin
+      - integration_tests_1
+      - integration_tests_2
+    with:
+      replicas: 1
+      account: 454262938596
+      workspace: integration
+    secrets: inherit
+
+  terraform_apply_preproduction:
+    name: environment apply terraform
+    uses: ./.github/workflows/_run-terraform.yml
+    needs:
+      - terraform_apply_shared_preproduction
+      - workflow_variables
+      - integration_tests_sequential
+      - integration_tests_admin
+      - integration_tests_1
+      - integration_tests_2
+    with:
+      workspace: preproduction
+      terraform_path: environment
+      apply: true
+      container_version: ${{ needs.workflow_variables.outputs.parsed_branch }}-${{ needs.workflow_variables.outputs.short_sha }}
+    secrets: inherit
+
+  terraform_apply_shared_production:
+    name: shared plan terraform development
+    uses: ./.github/workflows/_run-terraform.yml
+    needs:
+      - workflow_variables
+      - terraform_apply_preproduction
+    with:
+      workspace: production
+      terraform_path: shared
+      container_version: ${{ needs.workflow_variables.outputs.parsed_branch }}-${{ needs.workflow_variables.outputs.short_sha }}
+    secrets: inherit
+
+  terraform_apply_training:
+    name: environment apply terraform
+    uses: ./.github/workflows/_run-terraform.yml
+    needs:
+      - terraform_apply_shared_production
+      - workflow_variables
+    with:
+      workspace: training
+      terraform_path: environment
+      apply: true
+      container_version: ${{ needs.workflow_variables.outputs.parsed_branch }}-${{ needs.workflow_variables.outputs.short_sha }}
+    secrets: inherit
+
+  terraform_apply_production:
+    name: environment apply terraform
+    uses: ./.github/workflows/_run-terraform.yml
+    needs:
+      - terraform_apply_shared_production
+      - workflow_variables
+    with:
+      workspace: production02
+      terraform_path: environment
+      apply: true
+      container_version: ${{ needs.workflow_variables.outputs.parsed_branch }}-${{ needs.workflow_variables.outputs.short_sha }}
+    secrets: inherit
+
+  backup_production:
+    name: integration tests frontend 2
+    uses: ./.github/workflows/_run-task.yml
+    needs:
+      - workflow_variables
+      - terraform_apply_production
+    with:
+      workspace: production02
+      task_name: "backup"
+      timeout: "700"
+    secrets: inherit
+
+  restore_to_preproduction:
+    name: integration tests frontend 2
+    uses: ./.github/workflows/_run-task.yml
+    needs:
+      - workflow_variables
+      - terraform_apply_production
+    with:
+      workspace: preproduction
+      task_name: "restore_from_production"
+      timeout: "700"
+    secrets: inherit
+
+  end_of_workflow:
+    name: end of workflow
+    runs-on: ubuntu-latest
+    needs:
+      - restore_to_preproduction
+      - codecov
+      - workflow_variables
+    steps:
+      - name: workflow has ended without issue
```

Comment:
Minor thing. I'd be tempted to change this to "workflow ended succesfully", the "without issue" sounds a bit negative.

Reply:
Hehe, ok sold :)